### PR TITLE
Scalars for extension dtypes.

### PR DESCRIPTION
**Pull request by TomAugspurger**

### CODE REVIEW

**Comment by mrocklin** [comment]
@TomAugspurger what's the status on this?  Are you still working here?  Is this ready for people to review?

**Comment by TomAugspurger** [comment]
Fixed the error failing CI. I'll need to look a bit more closely again before saying whether this is ready. I don't recall where I was when I dropped it.

**Comment by TomAugspurger** [comment]
I've merged master to give the CI another run. I think this is ready to go, if you have a chance to look this over @jrbourbeau.

**Comment by jrbourbeau** [comment]
Thanks for the update @TomAugspurger, I'll make time tomorrow to review this

File: dask/dataframe/utils.py [hunk 1]
```diff
@@ -445,15 +442,33 @@ def _scalar_from_dtype(dtype):
...
 
 
+@make_scalar.register(np.dtype)
+def _(dtype):
+    return _scalar_from_dtype(dtype)
+
+
+@make_scalar.register(pd.Timestamp)
+@make_scalar.register(pd.Timedelta)
+@make_scalar.register(pd.Period)
```
Comment (jrbourbeau) [inline dask/dataframe/utils.py]: Does it make sense for `make_scalar` to handle `pd.Categorical` types? Just thinking about the extensions built into pandas

File: dask/dataframe/utils.py [hunk 2]
```diff
@@ -16,12 +16,9 @@
 from pandas.api.types import (is_categorical_dtype, is_scalar, is_sparse,
                               is_period_dtype, is_datetime64tz_dtype)
 
-try:
-    from pandas.api.types import is_interval_dtype
-except ImportError:
-    is_interval_dtype = lambda dtype: False
+from pandas.api.types import is_interval_dtype
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply dask/dataframe/utils.py]: Yes, is_interval_dtype / Interval is included in pandas 0.21.

File: dask/dataframe/utils.py (hunk 1 above)
Comment (TomAugspurger) [reply dask/dataframe/utils.py]: Categorical doesn't have a single scalar type like Timestamp or Period. It's a container for any scalar type, so there's no single type to register.

Comment (jrbourbeau) [reply dask/dataframe/utils.py]: Okay, thanks for clarifying

**Comment by jrbourbeau** [comment]
Merging on green

**Pull request merged**